src/repeater/repeater.js

```diff
@@ -90,47 +124,57 @@ function onFailure(err) {
 function stopAllRepeat() {
   debug('Entered repeater.stopAllRepeat');
   Object.keys(tracker).forEach((key) => {
-    if (tracker[key].stop) {
-      tracker[key].stop();
-    } else {
-      Object.keys(tracker[key]).forEach((nestedKey) => {
-        if (tracker[key][nestedKey].stop) {
-          tracker[key][nestedKey].stop();
-        }
-      });
-    }
-
-    delete tracker[key];
+    stop(key);
   });
 
   return tracker;
 } // stopAllRepeat
 
 /**
- * Stops the named repeater and deletes it from the tracker.
- *
- * @param {String} name - Name of the repeat
- * @throws {ValidationError} If "obj" does not have a name attribute.
- * @throws {ResourceNotFoundError} If the repeat identified by obj.name is not
- * found in the tracker.
+ * Pauses the repeater, given its name
+ * @param  {String} name - Name of the repeat
  */
-function stop(name) {
-  if (!name || !tracker[name]) {
-    throw new errors.ResourceNotFoundError(`Repeater "${name}" not found`);
-  }
+function pause(name) {
+  changeRepeatState(name, 'pause');
+  logger.info(`Paused the repeater identified by: ${name}`);
+} // pause
 
-  if (tracker[name].stop) {
-    tracker[name].stop();
-  } else {
-    Object.keys(tracker[name]).forEach((prop) => {
-      tracker[name][prop].stop();
-      delete tracker[name][prop];
-    });
-  }
+/**
+ * Pauses all the generator repeates.
+ * @returns {Object} The tracker object tracking all the repeats
+ */
+function pauseGenerators() {
+  Object.keys(tracker).forEach((key) => {
+    if (key !== 'heartbeat') {
```

Comment:
I don't love this hard-coded naming inequality test here... the intent seems to be that we only operate on "generator" trackers and not the "heartbeat" tracker... and we're using a naming convention to make that determination. I'd love to see a cleaner implementation which avoids reliance on mere naming convention.

If leaving this as is, I'd prefer to see it implemented using filter instead of if, i.e.

`Object.keys(tracker).filter((key) => key !== 'heartbeat').forEach(pause);`